plutus-ledger-api/src/Codec/CBOR/Extras.hs

```diff
@@ -20,3 +28,40 @@ decodeViaFlat decoder = do
     -- lift any flat's failures to be cborg failures (MonadFail)
     fromRightM (fail . show) $
         Flat.unflatWith decoder bs
+
+{- | The errors returned by `cborg` are plain strings (untyped). With this function we try to
+map onto datatypes, those cborg error messages that we are interested in.
+
+Currently we are only interested in error messages returned by the `CBOR.decodeBytes` decoder;
+see `PlutusLedgerApi.Common.SerialisedScript.scriptCBORDecoder`.
+-}
+interpretDeserialiseFailure :: CBOR.DeserialiseFailure -> DeserialiseFailureInterpreted
+interpretDeserialiseFailure (CBOR.DeserialiseFailure byteOffset reason) =
+    DeserialiseFailureInterpreted byteOffset $ interpretReason reason
+  where
+    -- Note that this is subject to change if `cborg` dependency changes. Currently: cborg-0.2.9.0
+    interpretReason :: String -> DeserialiseFailureReason
+    interpretReason = \case
+        -- Relevant Sources:
+        -- <https://github.com/well-typed/cborg/blob/cborg-0.2.9.0/cborg/src/Codec/CBOR/Read.hs#L226>
+        -- <https://github.com/well-typed/cborg/blob/cborg-0.2.9.0/cborg/src/Codec/CBOR/Read.hs#L1424>
+        -- <https://github.com/well-typed/cborg/blob/cborg-0.2.9.0/cborg/src/Codec/CBOR/Read.hs#L1441>
+        "end of input"   -> EndOfInput
+        -- Relevant Sources:
+        -- <https://github.com/well-typed/cborg/blob/cborg-0.2.9.0/cborg/src/Codec/CBOR/Read.hs#L1051>
+        "expected bytes" -> ExpectedBytes
+        msg              -> OtherReason msg
+
+data DeserialiseFailureInterpreted
```

Comment:
Haddock?

I think the "interpreted" here is not that useful for a consumer of this type. I think we could just say "DeserialiseFailureError" or something

Reply:
I don't like Interpreted, but I also don't like your suggestion DeserialiseFailureError  :D .

Reply:
So the thing I don't like is that "interpreted" is about how it was produced, not what it is.

Reply:
I turned it into `DeserialiseFailureInfo` and `readDeserialiseFailureInfo`. What do u think?

Reply:
Sure